components/ShareButton.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import styled from 'styled-components/native'
+import { Share } from 'react-native'
+
+import ShareIcon from './assets/share.png'
+
+const TouchableOpacity = styled.TouchableOpacity`
+  padding: 10px;
+`
+
+const Icon = styled.Image`
+  width: 35px;
+  height: 35px;
+`
+
+export const ShareButton = ({url}) => {
+  const onShare = async () => {
```

Comment:
Great job with `async await` :) For further explanation of this topic, you can check out https://medium.com/javascript-in-plain-english/async-await-javascript-5038668ec6eb